src/core/thread/mle.cpp

```diff
@@ -1724,7 +1726,11 @@ ThreadError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageI
     // Active Timestamp
     if (Tlv::GetTlv(aMessage, Tlv::kActiveTimestamp, sizeof(activeTimestamp), activeTimestamp) == kThreadError_None)
     {
+        const MeshCoP::Dataset local = mNetif.GetActiveDataset().GetLocal();
+        const MeshCoP::Timestamp *localTimestamp = local.GetTimestamp();
```

Comment:
Just call `mNetif.GetActiveDataset().GetLocal().GetTimestamp()` and avoid the allocation and copy on the stack.